lib/rules/object-property-newline.js

```diff
@@ -56,8 +56,8 @@ module.exports = {
                 }
 
                 for (var i = 1; i < node.properties.length; i++) {
-                    lastTokenOfPreviousValue = sourceCode.getLastToken(node.properties[i - 1].value);
-                    firstTokenOfCurrentKey = sourceCode.getFirstToken(node.properties[i].key);
+                    lastTokenOfPreviousValue = sourceCode.getLastToken(node.properties[i - 1]);
+                    firstTokenOfCurrentKey = sourceCode.getFirstToken(node.properties[i]);
```

Comment:
I think the variables should be renamed to reflect what they store now. How about something like:
`lastTokenOfPreviousProperty` and `firstTokenOfCurrentProperty`